src/edu/ucsb/nceas/metacat/dataone/D1AuthHelper.java

```diff
@@ -365,13 +365,24 @@ public void doAdminAuthorization(Session session) throws ServiceFailure, NotAuth
         // If subject is not a LocalNodeAdmin or CNAdmin, check to see if subject is a
         // Metacat admin (auth.administrators) - who also have admin privileges like the above
         try {
-            String adminUser = session.getSubject().getValue();
-            if (adminUser != null) {
-                logMetacat.debug("D1AuthHelper.doAdminAuthorization: Checking " + adminUser +
-                                     " for Metacat admin privileges.");
-                if (AuthUtil.isAdministrator(adminUser, null)) {
-                    return;
+            // If session is null, there is no user so admin is not authorized.
+            if (session != null) {
+                String adminUser = session.getSubject().getValue();
+                if (adminUser != null) {
+                    logMetacat.debug("D1AuthHelper.doAdminAuthorization: Checking " + adminUser
+                        + " for Metacat admin privileges.");
+                    if (adminUser.trim().isEmpty()) {
+                        throw new NotAuthorized(
+                            "0000", "Session is not null, but adminUser is empty.");
+                    }
+                    if (AuthUtil.isAdministrator(adminUser, null)) {
```

Comment:
if this statement return false, no exceptions will be thrown. So this method will authorize the session.

Reply:
I have added an else statement here to throw a `NotAuthorized` exception when the return value is false. Thank you for catching this.